src/otp-input/otp-input.tsx

```diff
@@ -0,0 +1,182 @@
+        event: React.KeyboardEvent<HTMLInputElement>,
+        index: number
+    ) => {
+        const { key, code } = event;
+        const validKey = [
+            "Backspace",
+            "Tab",
+            "ShiftLeft",
+            "ShiftRight",
+            "Shift",
+        ];
+        if (!validKey.includes(code) || !validKey.includes(key)) {
+            if (!RegExp(/^[0-9]+$/).test(key)) {
+                event.preventDefault();
+            }
+        }
+
+        const newOtpValues = [...otpValues];
+        if (
+            (code === "Backspace" || key === "Backspace") &&
+            newOtpValues[index] !== ""
+        ) {
+            newOtpValues[index] = "";
+        } else if (code === "Backspace" || key === "Backspace") {
+            newOtpValues[index - 1] = "";
+            inputRefs.current[index - 1]?.focus();
+        }
+
+        if (validateUserInput(key)) {
+            newOtpValues[index] = key;
+            inputRefs.current[index + 1]?.focus();
+        }
+
+        setOtpValues(newOtpValues);
+
+        if (onChange) {
+            onChange(newOtpValues.join(""));
+        }
+    };
+
+    const handlePaste = (event: ClipboardEvent): void => {
+        const pastedValue = event.clipboardData.getData("text");
+        if (pastedValue && validateUserInput(pastedValue, numOfInput)) {
+            setOtpValues(pastedValue.split(""));
+            if (onChange) {
+                onChange(pastedValue);
+            }
+        } else {
+            event.preventDefault();
+        }
+    };
+
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setOtpValues(new Array(numOfInput).fill(""));
+        setEnableCountDownTimer(true);
+        if (onClick) {
+            onClick(event);
+        }
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const validateUserInput = (value: string, length = 1) => {
+        if (!value) {
+            return false;
+        }
+        return RegExp(`^[0-9]{${length}}$`).test(value);
+    };
+
+    const renderCTALabel = () => {
+        return otherCtaProps.children
+            ? `${otherCtaProps.children}`
+            : `Resend OTP ${
+                  countDown
+                      ? `in ${countDown} second${countDown > 1 ? "s" : ""}`
+                      : ""
+              }`;
+    };
+
+    return (
+        <Wrapper>
+            <InputContainer>
+                {otpValues.map((data, index) => {
+                    return (
+                        <InputField
+                            key={index}
+                            id={"otpValues-input"}
+                            data-testid={"otpValues-input"}
+                            ref={(el) => (inputRefs.current[index] = el)}
+                            onKeyDown={(e) => handleKeyDown(e, index)}
```

Comment:
Suppose you curried the function then this would be suficient 
```suggestion
                            onKeyDown={handleKeyDown(index)}
```